vmdb/app/models/ems_refresh/save_inventory_helper.rb

```diff
@@ -99,4 +99,27 @@ def store_ids_for_new_records(records, hashes, keys)
       h[:id] = r.id
     end
   end
+
+  # most of the refresh_multi calls follow the same pattern
+  # this pulls it out
+  def save_inventory_assoc(type, parent, hashes, target, find_key, child_keys = [], extra_keys = [])
+    reflection = parent.class.reflect_on_association(type)
+    klass = reflection.class_name.constantize
+    deletes = relation_values(parent, reflection, target)
+
+    save_inventory_multi(type, klass, parent, hashes, deletes, find_key, child_keys, extra_keys)
+    store_ids_for_new_records(parent.send(type), hashes, find_key)
+  end
+
+  # We need to determine our intent:
+  # - make a complete refresh. Delete missing values.
+  # - make a partial refresh. Don't delete missing keys
+  # This generates the "deletes" values based upon this intent
+  def relation_values(parent, reflection, target)
+    # always want to refresh this association
+    reflection = parent.class.reflect_on_association(reflection) if reflection.kind_of?(Symbol)
+    top_level = reflection.options[:dependent] == :destroy
```

Comment:
I don't get the dependent => destroy check.  What does that have to do with updating an association?